File: src/main/java/noteedit/entity/FolderInfo.java
package noteedit.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class FolderInfo {
    private Long id;

    private String folderName;

    private Long father;
    @TableField(exist = false)
    private List<Long> children;
    @TableField("children")
    private String childrenJson;
    private Integer depth;
}

File: src/main/java/noteedit/entity/FolderInfo.kt
package noteedit.entity

import com.baomidou.mybatisplus.annotation.TableField
import noteedit.utils.JsonUtils

data class FolderInfo(
    var id: Long = 0, // 编号
    var name: String = "", // 文件夹名称
    var depth: Int = 0, // 层级
) {
    @TableField("father")
    var fatherId: Long = 0 // 父级编号
    @TableField(exist = false)
    var subfolders: MutableList<Long>? = null // 子文件夹
    @TableField("children")
    var subfolderJson: String? = "" // 子文件夹(json)
        get() = JsonUtils.objectToJson(subfolders)
        set(value) {
            if (value?.isNotBlank() == true) {
                subfolders = JsonUtils.jsonToObject<MutableList<Long>>(value)
                field = value
            }
        }
}